utility/session.go

```diff
@@ -1,157 +1,266 @@
+		if slices.Contains(servicer.Chains, s.session.RelayChain) {
+			candidateServicers = append(candidateServicers, servicer)
+		}
+	}
+
+	s.session.Servicers = pseudoRandomSelection(candidateServicers, numServicers, s.sessionIdBz)
+	return nil
 }
 
-// uses the current 'world state' to determine the servicers in the session
-// 1) get an ordered list of the public keys of servicers who are:
-//   - actively staked
-//   - staked within geo-zone (or closest geo-zones)
-//   - staked for relay-chain
-//
-// 2) calls `pseudoRandomSelection(servicers, numberOfNodesPerSession)`
-func (s *session) selectSessionServicers(numServicers int) ([]*coreTypes.Actor, types.Error) {
-	// IMPORTANT: This function is for behaviour illustrative purposes only and implementation may differ.
-	return nil, nil
+// hydrateSessionFishermen finds the fishermen that are staked at the session height and populates the session with them
+func (s *sessionHydrator) hydrateSessionFishermen() error {
+	// number of fishermen per session at this height
+	numFishermen, err := s.readCtx.GetIntParam(types.FishermanPerSessionParamName, s.session.SessionHeight)
+	if err != nil {
+		return err
+	}
+
+	// returns all the staked fishermen at this session height
+	fishermen, err := s.readCtx.GetAllFishermen(s.session.SessionHeight)
+	if err != nil {
+		return err
+	}
+
+	// OPTIMIZE: Consider updating the persistence module so a single SQL query can retrieve all of the actors at once.
+	candidateFishermen := make([]*coreTypes.Actor, 0)
+	for _, fisher := range fishermen {
+		// Sanity check the fisher is not paused, jailed or unstaking
+		if !(fisher.PausedHeight == -1 && fisher.UnstakingHeight == -1) {
+			return fmt.Errorf("hydrateSessionFishermen should not have encountered a paused or unstaking fisherman: %s", fisher.Address)
+		}
+
+		// TODO(#697): Filter by geo-zone
+
+		// OPTIMIZE: If this was a map[string]struct{}, we could have avoided the loop
+		if slices.Contains(fisher.Chains, s.session.RelayChain) {
+			candidateFishermen = append(candidateFishermen, fisher)
+		}
+	}
+
+	s.session.Fishermen = pseudoRandomSelection(candidateFishermen, numFishermen, s.sessionIdBz)
+	return nil
 }
 
-// uses the current 'world state' to determine the fishermen in the session
-// 1) get an ordered list of the public keys of fishermen who are:
-//   - actively staked
-//   - staked within geo-zone  (or closest geo-zones)
-//   - staked for relay-chain
-//
-// 2) calls `pseudoRandomSelection(fishermen, numberOfFishPerSession)`
-func (s *session) selectSessionFishermen(numFishermen int) ([]*coreTypes.Actor, types.Error) {
-	// IMPORTANT: This function is for behaviour illustrative purposes only and implementation may differ.
-	return nil, nil
+// pseudoRandomSelection returns a random subset of the candidates.
+// DECIDE: We are using a `Go` native implementation for a pseudo-random number generator. In order
+// for it to be language agnostic, a general purpose algorithm needs ot be used.
```

Comment:
Nice call out! :100: 

super nit: Wdyt about adopting [rfc2119 style requirement keywords](https://www.rfc-editor.org/rfc/rfc2119) (eg. `MUST`, `SHOULD`, `MAY`, `... NOT`)
```suggestion
// DECIDE: We are using a `Go` native implementation for a pseudo-random number generator. In order
// for it to be language agnostic, a general purpose algorithm MUST be used.
```

---

Alternatively there's a typo  - I think that's what @dylanlott was commenting on:
```suggestion
// DECIDE: We are using a `Go` native implementation for a pseudo-random number generator. In order
// for it to be language agnostic, a general purpose algorithm needs to be used.
```